app/electron/main.ts

```diff
@@ -453,6 +453,42 @@ class PluginManagerEventListeners {
   }
 }
 
+function getShellEnv(): NodeJS.ProcessEnv {
+  if (process.platform === 'win32') {
+    return process.env;
+  }
+
+  let shell = process.env.SHELL || '/bin/sh';
+  if (process.platform === 'darwin') {
+    try {
+      shell = userInfo().shell || '/bin/zsh';
+    } catch (error) {
+      shell = '/bin/zsh';
+    }
+  }
+
+  try {
+    const env = { ...process.env, DISABLE_AUTO_UPDATE: 'true' };
+    const result = execSync(`${shell} -ilc 'env'`, {
```

Comment:
When running zsh, in the unlikely but possible case the user has no config... then zsh can put up an interactive mode telling the user which causes zsh to hang.
To disable this the `--no-rcs` can be used (on zsh only).

```
          .concat(shell === '/bin/zsh' ? [] : ['--no-rcs'])
```

Reply:
adding `--no-rcs` is breaking the solution, in my case i have the required configurations setup in the .zshrc file.

Reply:
I checked it by removing my .zshrc and running the command. zsh doesn't get into informing the user to setup the config.
<img width="550" alt="image" src="https://github.com/user-attachments/assets/5a259e57-5b9d-45c3-9e38-efd3affe107c">